src/policy/markcompactspace.rs

```diff
@@ -51,6 +52,18 @@ impl<VM: VMBinding> SFT for MarkCompactSpace<VM> {
     fn is_sane(&self) -> bool {
         true
     }
+
+    #[inline(always)]
+    fn sft_trace_object(
+        &self,
+        _trace: SFTProcessEdgesMutRef,
+        _object: ObjectReference,
+        _worker: GCWorkerMutRef,
+    ) -> ObjectReference {
+        // We should not use trace_object for markcompact space.
+        // Depending on which trace it is, we should manually call either trace_mark or trace_forward.
+        panic!("sft_trace_object() cannot be used with mark compact space")
```

Comment:
It is possible to define a enum variable `trace_kind` in mark compact plan, indication which trace it is doing (should be either "mark" or "forward"). Then we can do a branch here to either `trace_mark` or `trace_forward`, based on the trace state.

Reply:
This may simplify the code (?) since we don't need to define other `ProcessEdgesWork` structs in markcompact code for tracing.

Reply:
I gave it a try. Theoretically it could work. But we would need to set `trace_kind` 4 times: before tracing, before forwarding, before tracing finalizable objects, before forward finalizable objects. And we do not have a proper hook for the last two (we do not have a hook to run some code when a work bucket is opened) so we will probably end up having more code. 

I could work on it after this pull request, and possibly add a hook in the scheduler to call some methods in the plan when a work bucket is first opened. Let me know what you think.

Another concern is performance, as this requires us to do a branch for each `trace_object` call.